### DISCUSSION TITLE
UI: Open musician profile on click on own slider?

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1027] [UI Related]
**Is your feature request related to a problem? Please describe.**

New users might find it difficult to find the profile button (just had someone yesterday who didn't find it). 

**Describe the solution you'd like**

Add a button (pencil or similar) on the own slider next to the name which opens the musician profile window.

**Describe alternatives you've considered**

* Add a button to open the profile somewhere visible

**Additional context**

Related to my findings where the UI could be improved.

**Comment by pljones** [comment]
That was the one big loss with the "Mute myself" checkbox going where it did - the profile button fitted there nicely.

**Comment by gilgongo** [comment]
Having to scatter navigation and controls around the UI like this will become more of a problem as time goes on without some intervention on the structure of the interface overall.  This was my point 3 here https://github.com/jamulussoftware/jamulus/issues/958#issuecomment-778585529

I think doing this ticket would be OK for now, but it's storing up design debt really.

**Comment by ghost** [comment]
"Open musician profile on click on own slider" may address a user proplem, but is not the solution because:
- your own slider is only there when you are connected to a server. (an easter egg)
- someone who cannot find (View > My Profile ) has no vested interest in using Jamulus.

**Comment by dcorson-ticino-com** [comment]
>Having to scatter navigation and controls around the UI

These things belong under Settings with a multi-tabbed window, User, Network, Device ....  or similar

**Comment by pljones** [comment]
There was a back burner topic about moving to a multi-tabbed design.  The Server UI got one step in that direction.  I think that's probably the best approach here.  Gather up _all_ the settings in to one place, then arrange them into logical groups in tabs, as you suggest.

**Comment by dcorson-ticino-com** [comment]
I will look into doing this in the next week or so.
And at the same time clean up the menus #994

**Comment by ann0see** [comment]
Thank you very much for this work!

**Comment by gilgongo** [comment]
Given the fact that there is now, for better or worse, a convention of having navigation in most apps on a bar on the left, it would be good to consider that for Jamulus in order to present a familiar environment and so that we can scale the UI for the future if needed.

**Comment by dcorson-ticino-com** [comment]
>a convention of having navigation in most apps on a bar on the left

Is this true?  I am not aware of any program like that.
And even if it is true I don't think we need to run after the latest fashion.
The simple act of moving the Connect button from the far left of the window to the middle makes Jamulus IMHO much more convivial, easier and quicker to use.
Also for a vertical menu text descriptions must be very short (or none), otherwise the menu becomes very wide.
When I think about it the only place I use vertical menus in a program is in CAD toolbars where there are many tools and toolbars and they have only icons, no text.

(I understand why people would put the menus etc at the left or right edge, as the screens are no longer sized to be a good work environment, but to show movies and as such they are way too wide.   But pushing the menus to the dead portions of the screen makes them dead too. Personally I have my task bar at the right and not at the bottom as is standard, but that is because I rarely use it.)

**Comment by gilgongo** [comment]
It's [mostly true](https://www.google.com/search?q=UI+navigation+examples+desktop+web&tbm=isch&ved=2ahUKEwjRtZezjOnuAhXE0aQKHYQGCWUQ2-cCegQIABAA&oq=UI+navigation+examples+desktop+web&gs_lcp=CgNpbWcQA1CGpwFYm6sBYKuvAWgAcAB4AIABXogB1QKSAQE0mAEAoAEBqgELZ3dzLXdpei1pbWfAAQE&sclient=img&ei=_PEoYNHHCcSjkwWEjaSoBg&bih=665&biw=1366&rlz=1CAWPBA_enGB904). And indeed you gave an example earlier with Windows. I think we agrees that you wanted tool tips.

In general, the adage here is that, "All of your users spend most of their time with other people's applications". If you want to help orientate people, it's usually a good idea to start with a navigation scheme they are likely to be familiar with.

But let's see what you come up with.

**Comment by dcorson-ticino-com** [comment]
>But let's see what you come up with.

In the immediate, nothing special, just a tabbed window with all the settings.

For the NewUI I have ideas, have to try out some stuff, will take time.

**Comment by ann0see** [comment]
> I will look into doing this in the next week or so.

Assigned you to this issue (I'm doing a few tests for project management features in GitHub now. I hope it's ok for you to be assigned here)

**Comment by Hk1020** [comment]
> > a convention of having navigation in most apps on a bar on the left
> 
> Is this true? I am not aware of any program like that.
> And even if it is true I don't think we need to run after the latest fashion.
> The simple act of moving the Connect button from the far left of the window to the middle makes Jamulus IMHO much more convivial, easier and quicker to use.
> Also for a vertical menu text descriptions must be very short (or none), otherwise the menu becomes very wide.
> When I think about it the only place I use vertical menus in a program is in CAD toolbars where there are many tools and toolbars and they have only icons, no text.
> 
> (I understand why people would put the menus etc at the left or right edge, as the screens are no longer sized to be a good work environment, but to show movies and as such they are way too wide. But pushing the menus to the dead portions of the screen makes them dead too. Personally I have my task bar at the right and not at the bottom as is standard, but that is because I rarely use it.)



> > a convention of having navigation in most apps on a bar on the left
> 
> Is this true? I am not aware of any program like that.

Two that come to mind - Microsoft Teams (very popular these days), Visual Studio Code.

**Comment by pljones** [comment]
> Two that come to mind - Microsoft Teams (very popular these days), Visual Studio Code.

I nearly said I couldn't think of any.  Then I remembered - all the IDEs I use work like that: document in the centre, navigation and tools around the outside.   Also, desktop mail applications, calendar applications, etc.  And mobile apps with left hand pull-out menus....

Is the "document" for Jamulus the session?  What count as tools?  All my IDEs still have menus, as well as tool bars and navigation facilities, status windows, additional information, etc.

The _biggest_ difference - it's *huge* - is that almost every other application I use has all it's content in a single window, often tabbed if multi-document (with the option of tabs in separate windows).

**Comment by dcorson-ticino-com** [comment]
>I nearly said I couldn't think of any. Then I remembered - all the IDEs

Well, I looked again, I do use one program that has a left menu, Qt Designer.
The UI is only usable for developers I'd say.
There is a left menu, but I have never used it.

**Comment by gilgongo** [comment]
> The UI is only usable for developers I'd say.

That will be news to Facebook, YouTube, WhatsApp and pretty much all others :-)

**Comment by dcorson-ticino-com** [comment]
>The UI is only usable for developers I'd say.

I'm talking about Qt Designer, microscopic non-standard icons, windows over windows that open automagically for no apparent reason.  Then the hunt through the weird icons to find how to close them.  Impossibility to do normal things (show two *different* files next to each other may be possible, but I still haven't found it if it is.  I can open a multitude of windows with the same file however.)  As I say SW developers are willing to accept a lot just to become part of the gang.  The useless menu on the left of the screen is not one of the programs problems.  And the functionality of the program is pretty good so no-one complains about the UI.  Why do people complain about the UI of Jamulus, the functionality is great?!

**Comment by dakhubgit** [comment]
> "Open musician profile on click on own slider" may address a user proplem, but is not the solution because:
> 
>     * your own slider is only there when you are connected to a server. (an easter egg)

Well, that's something that should be addressed: there is no point in hiding one's slider until one is connected.  Making the profile accessible from the slider also would fit nicely in with a feature that allows having (and exporting) more than one slider, like one slider for my voice and one slider for my piano.  "Just start another Jamulus" session is not really the same since another Jamulus session also occupies another downstream connection needlessly, and Jamulus does not even allow (yet) to configure differently up-and-downstream quality, stereosity (mono-in/stereo-out needlessly sends in stereo), and presence (what use has a streaming client of an upstream?  What use a separate instrument of a downstream?).

Allowing to add sliders while offline and give them a profile (on the fly or by selecting one of several previously configured ones) would certainly be beneficial.

>     * someone who cannot find (View > My Profile ) has no vested interest in using Jamulus.

I don't buy the elitist "if our software is too hard/inconvient to use, it saves us from dealing with weak people" argument.  Nobody uses only a single piece of software and there is no point of making software hard to use gratuitously.  Using software should be fun rather than building character.

**Comment by ghost** [comment]
Jamulus is currently so simple that one can find what they want by spending less than 5 minutes to try all menus. Thus, if someone cannot find (View > My Profile ), they had not bothered to spend 5 minutes to change their Profile before suggesting how to do it differently.
NO more menus please. This creates clutter that will make it difficult for everyone (never mind new users).
Jamulus (and just about all else) cannot be made idiot proof, so don't waste effort making things even more compilcated for all. Yes, the changes you want will make Jamulus harder to use for the unintiated too.

**Comment by ghost** [comment]
@dcorson-ticino-com 
 Why do people complain about the UI of Jamulus, the functionality is great?!

Hear! Hear!
David

**Comment by gilgongo** [comment]
> Why do people complain about the UI of Jamulus, the functionality is great?!

It is now, but as I've said https://github.com/jamulussoftware/jamulus/issues/988#issuecomment-778605008, the design of a UI is not all about usability. It's also about **scalability**. Those are two separate but related things. It's also why most apps have adopted (for example), single window, multi-panelled structures as it allow clear hierarchic relationships and large feature sets within a predictable pattern. I think this ticket is therefore dependent on that.

**Comment by pljones** [comment]
> there is no point in hiding one's slider until one is connected.

There is no functional use in _any of the sliders_ except when connected to a server.  They set the mix at the server.  If you're not connected to a server, there's no use showing them.  I've no idea what the term "Easter egg" is referring to in this context.  To have the slider visible when non-functional would be misleading.

**Comment by gilgongo** [comment]
> slider visible when non-functional would be misleading

There is usually the principle of showing everything but then disabling (and explaining why, usually with a tooltip) those parts that are logically disabled. The thinking behind that is so that it's easier to learn/remember where you would find those functions in the enabled state. But it depends a bit on the design.

**Comment by dakhubgit** [comment]
Peter L Jones <notifications@github.com> writes:

>> there is no point in hiding one's slider until one is connected.
>
> There is no functional use in _any of the sliders_ except when
> connected to a server.

Factually wrong since the sliders work fine when using "Mute myself at
server" is selected.

> They set the mix at the server.  If you're not connected to a server,
> there's no use showing them.

Figuring out your settings before setting up howling and screeching for
everyone?

> I've no idea what the term "Easter egg" is referring to in this
> context.  To have the slider visible when non-functional would be
> misleading.

To have to connect to a server (and announce your presence in the
process and potentially disrupt a session and/or get recorded) just in
order to figure out your local settings and get acquainted with the
application is annoying.

-- 
David Kastrup

**Comment by gilgongo** [comment]
> To have to connect to a server ... in order to figure out your local settings and get acquainted with the application is annoying.

Yes, although it has been suggested https://github.com/jamulussoftware/jamulus/issues/882 that we have at a "loopback test" function instead (and meanwhile, you can always connect to an empty server, but of course that may not occur to to new users if they don't think they have problems in the first place)

**Comment by dakhubgit** [comment]
Jonathan <notifications@github.com> writes:

>> To have to connect to a server ... in order to figure out your local
> settings and get acquainted with the application is annoying.
>
> Yes, although it has been suggested
> https://github.com/jamulussoftware/jamulus/issues/882 that we have at
> a "loopback test" function instead (and meanwhile, you can always
> connect to an empty server,

It's still treading on someone else's ground, and once you are connected
to an "empty" public server, you have about 1 minute of experimentation
before someone joins in the hope of finding a jam session.

Been there, done that.

> but of course that may not occur to to new users if they don't think
> they have problems in the first place)

-- 
David Kastrup

**Comment by dcorson-ticino-com** [comment]
@dakhubgit 
>Factually wrong since the sliders work fine when using "Mute myself at
server" is selected.

Not really, it does what you say when you are connected to a server, when not connected you hear nothing.

Set "Mute Yourself" before connecting and do you setup when connected as no-one will hear you, is the way it works.
Do not confuse the Ist Zustand with wishes for the future.

**Comment by dakhubgit** [comment]
> @dakhubgit
> 
> > Factually wrong since the sliders work fine when using "Mute myself at
> > server" is selected.
> 
> Not really, it does what you say when you are connected to a server, when not connected you hear nothing.

I was countering the statement that operation of the sliders depends on being connected to the server.  The _user interface_ may demand you are connected to a server even in "Mute yourself" mode, but the actual data flow then bypasses the server.  There is no point served in differentiating your own slider operation and visibility in "Mute yourself" mode depending on whether or not you are actually connected to a server.

> Set "Mute Yourself" before connecting and do you setup when connected as no-one will hear you, is the way it works.
> Do not confuse the Ist Zustand with wishes for the future.

I was describing the situation connected to a server with "Mute yourself" activated.  In that case, your own sliders are implemented to operate in a manner that would perfectly well continue to function if the server dropped the connection.  It's only when your client notices that the server is dead that your sliders become non-functional.

So the normal "offline mode" could probably be made equivalent to a combination of "Mute yourself" and connecting to a non-existent server (which times out indefinitely).

**Comment by dcorson-ticino-com** [comment]
@dakhubgit 
Have you actually tried the scenarios you describe?
There is no data flow bypassing the server.
Even in "Mute Yourself" the data goes to the server and back to you, but to no-one else.
It is only because of that behaviour that mute yourself is useful for setup and testing.
But if you are not connected to a server you will hear nothing, no matter what you do.

**Comment by dakhubgit** [comment]
> @dakhubgit
> Have you actually tried the scenarios you describe?
> There is no data flow bypassing the server.
> Even in "Mute Yourself" the data goes to the server and back to you, but to no-one else.
~~~
    {
        // OPUS encoding
        if ( CurOpusEncoder != nullptr )
        {
            if ( bMuteOutStream )
            {
                iCodedSize = opus_custom_encode ( CurOpusEncoder,
                                                  &vecZeros[i * iNumAudioChannels * iOPUSFrameSizeSamples],
                                                  iOPUSFrameSizeSamples,
                                                  &vecCeltData[0],
                                                  iCeltNumCodedBytes );
            }
            else
            {
                iCodedSize = opus_custom_encode ( CurOpusEncoder,
                                                  &vecsStereoSndCrd[i * iNumAudioChannels * iOPUSFrameSizeSamples],
                                                  iOPUSFrameSizeSamples,
                                                  &vecCeltData[0],
                                                  iCeltNumCodedBytes );
            }
        }

        // send coded audio through the network
        Channel.PrepAndSendPacket ( &Socket,
                                    vecCeltData,
                                    iCodedSize );
    }
~~~

**Comment by pljones** [comment]
> There is usually the principle of showing everything but then disabling

We should show _every user currently connected to any server anywhere_ then with their slider disabled?  That's not helpful.

> Factually wrong since the sliders work fine when using "Mute myself at server" is selected.

Um?  How do you mute yourself at the server if you're not connected?

> Figuring out your settings before setting up howling and screeching for everyone?

Different issue.

> To have to connect to a server (and announce your presence in the process and potentially disrupt a session and/or get recorded) just in order to figure out your local settings and get acquainted with the application is annoying.

This means "Easter egg"?  Sorry, I wasn't aware of that usage of the term.

**Comment by dakhubgit** [comment]
> > Factually wrong since the sliders work fine when using "Mute myself at server" is selected.
> 
> Um? How do you mute yourself at the server if you're not connected?

I was countering the statement that operation of the sliders depends on being connected to the server. The user interface may demand you are connected to a server even in "Mute yourself" mode, but the actual data flow then bypasses the server. There is no point served in differentiating your own slider operation and visibility in "Mute yourself" mode depending on whether or not you are actually connected to a server.

**Comment by pljones** [comment]
OK, so you're saying if you've chosen "Mute Myself", display the user's local slider.  That does have some logic -- except for the fact that there's no audio process running.  So they wouldn't hear themselves.

**Comment by dakhubgit** [comment]
> OK, so you're saying if you've chosen "Mute Myself", display the user's local slider. That does have some logic -- except for the fact that there's no audio process running. So they wouldn't hear themselves.

No, I am saying that the operation of "Mute Myself" proves that the server is not required for letting your channel strip operate and produce output hearing yourself.  So it is little more than chicanery that your channel strip only becomes visible and operative once you are connected to a server.  I also rather doubt your assertion "there's no audio process running" since client's and server's timing are not synchronised and the client refuses to even start up (in the Linux version) without a sound system providing the timing.

**Comment by dcorson-ticino-com** [comment]
>No, I am saying that the operation of "Mute Myself" proves that the server is not required for letting your channel strip operate and produce output hearing yourself.

David, you still haven't answered my question.
Have you put your headphones on, started Jamulus and made sounds that you hear with your headphones, and then tried the scenarios you are talking about?
If you do you will find that without connecting to a server you never hear anything, with or without "Mute Yourself"
Take the time to try it.

**Comment by dakhubgit** [comment]
> > No, I am saying that the operation of "Mute Myself" proves that the server is not required for letting your channel strip operate and produce output hearing yourself.
> 
> David, you still haven't answered my question.
> Have you put your headphones on, started Jamulus and made sounds that you hear with your headphones, and then tried the scenarios you are talking about?
> If you do you will find that without connecting to a server you never hear anything, with or without "Mute Yourself"
> Take the time to try it.

The "Mute Yourself" functionality proves that there is no necessity of bouncing the sound through a server for hearing yourself and having an operative strip and that the current user interface blocks you from testing things without connection entirely gratuitously.  If the user interface didn't, we would not be having this discussion in the first place.

What you can do is connect to a server, then cut your network connectivity (take out the cable, switch off the WiFi, whatever).  While Jamulus hasn't yet given up on the no longer reachable server, you'll see your mixer strip and hear yourself in your headphones _when and only when_ "Mute yourself" is active, clearly without any involvement of the remote server that has become unreachable.  After about 10 seconds Jamulus decides the server is gone for good and reverts to being useless: it removes your mixer strip and does not allow you to test its effects.

**Comment by pljones** [comment]
> I also rather doubt your assertion "there's no audio process running" since client's and server's timing are not synchronised and the client refuses to even start up (in the Linux version) without a sound system providing the timing.

> What you can do is connect to a server, then cut your network connectivity (take out the cable, switch off the WiFi, whatever). While Jamulus hasn't yet given up on the no longer reachable server

That's two different things.

I said when the audio process is not running.  You've demonstrated that the audio process keeps trying - due to using UDP, it needs to - when there's a network brown out.

**Comment by dakhubgit** [comment]
But the metering works.  At least it used to until the following commit was added just now to master:
~~~
commit a7ecdf4b26f5f774a5f2e32585949b644c64bf3b
Author: ann0see <20726856+ann0see@users.noreply.github.com>
Date:   Fri Feb 12 22:33:13 2021 +0100

    Disable input level meters if not connected
~~~
It would appear there is a mission to make Jamulus as useless as possible when not connected to a server.  Frankly, I don't get the point.  Is this trying to prove something?

**Comment by pljones** [comment]
```
@@ -1239,6 +1242,8 @@ void CClientDlg::Disconnect()

     // stop timer for level meter bars and reset them
     TimerSigMet.stop();
+    lbrInputLevelL->setEnabled ( false );
+    lbrInputLevelR->setEnabled ( false );
     lbrInputLevelL->SetValue ( 0 );
     lbrInputLevelR->SetValue ( 0 );
```
Note this:
```
     TimerSigMet.stop();
```
which was already there.  That stop processing anything in the UI when Disconnect() is called.  All the commit you refer to does is set the UI status to "disabled" - it doesn't change function.

**Comment by dakhubgit** [comment]
There is a philosophy at work here that I don't get.  It may be related to different operating systems: on GNU/Linux, the Jack sound system needs to be up and running to even start Jamulus, and that means that in general everything is already set up.  There is no way inside of Jamulus to choose different input ports, a different sound device, Midi ports for a controller or whatever.  So the audio system is always engaged and ready and there is no point in not making use of it.

It's conceivable that this is not similarly the case on other platforms.

However, it is borderline annoying how inflexible Jamulus is regarding the sound system on Linux: it takes the first two physical input ports without asking questions.  There is an input selector box that has only one entry: "Jack".

Personally, it would make a lot more sense if local slider(s) were present and operative independent of server and you could click on the slider(s) to connect them to particular inputs.  Similarly for the output.  Having everything disappear and inoperative without a server serves no useful purpose that I can see.

**Comment by gilgongo** [comment]
This ticket seems to have forked into different related discussions, so I'm renaming it and moving it there. The original feature request may still be valid to add to the issues list later once we have some agreement on how to do that. 

And if anyone has any actionable, defined feature requests on the back of this conversation, please raise them as clearly specced tickets we can add to the backlog. Otherwise, please open in Discussions and we can firm them up. 

![bitmoji](https://sdk.bitmoji.com/render/panel/456e2fde-22a9-4f4f-b5c1-033d5d15ef7e-e453f381-f13d-466a-a20e-c03c786b333b-v1.png?transparent=1&palette=1&width=246)

**Comment by ann0see** [comment]
After the move this discussion became quite inactive ;-). But I think it has very valuable input:

1. It seems to be confusing that the UI looks almost the same if you're not connected and Jamulus is "useless" in this state. Also there's no concensus what to do in disconnected state. 
a) Either Jamulus should give back sound (local monitoring) --> Probably possible, but requires some code change or 
b) We e.g. show the server list in disconnected state.
1. Currently you can't start Jamulus if there are audio-interface issues (wrong sample rate, jack not running,...) which is "rather unhelpful" (quote from pljones) in another issue. @dakhubgit I think that's one of your (and also my) points here?

**Reply by gilgongo** [reply]
b) as was sort of explored in one of my UIs. Not sure what the value of a) would really be.

BTW has this got anything to do with the name of this discussion? I'm not complaining - it's just a discussion - but I was curious in case there was a connection I'd missed.

**Reply by dakhubgit** [reply]
I think showing the server list in disconnected state in the mixerboard does not make a lot of sense since the windows are completely different in layout and some mixerboard elements (like the balance and reverb controls) are accessible even when not connected.  I don't see why it should be "confusing that the UI looks almost the same if you're not connected" when the same functionality is available.  If yours is the only slider, it should not really make a difference whether or not there is a server invoived.

I brought up point 2 as an argument that it's particularly pointless to have the audio switched off by default when it has to be operative to even start Jamulus.

**Reply by ann0see** [reply]
> BTW has this got anything to do with the name of this discussion?

No. But the discussion went into a totally different direction. But that's ok since it led to a lot of valuable input concerning UI.

**Reply by ann0see** [reply]
> I brought up point 2 as an argument that it's particularly pointless to have the audio switched off by default when it has to be operative to even start Jamulus.

I think a lot of people agree on this point. 

It should be possible to hear sound even if you're not connected but also clear that you need to connect to a server to jam with others. The question here would be: how can we make this clear?

**Reply by dakhubgit** [reply]
> b) as was sort of explored in one of my UIs. Not sure what the value of a) would really be.

The value is that you can try out the application in "dry run" mode without requiring to set up a server of your own or trespass on someone else's server for even the most basic first steps.

For example, you have no way to figure out whether your slider has a sensible name and instrument symbol without advertising it on some server.

> BTW has this got anything to do with the name of this discussion? I'm not complaining - it's just a discussion - but I was curious in case there was a connection I'd missed.

The connection with the discussion was tangential: in case of another different feature, namely being able to use a single client for transmitting several channels, you'd want an obvious ability to give your own sliders separate different profiles.  Being able to click on them for entering stuff would be easily discoverable (actually also when only having a single slider: took me some time till I figured out where to configure it and it did not help that under XFCE4, Qt chose not to offer the respective menus at all).

**Reply by ann0see** [reply]
> The value is that you can try out the application in "dry run" mode without requiring to set up a server of your own or trespass on someone else's server for even the most basic first steps.

Yep! This would solve a lot of problems also concerning the ASIO4ALL setup. I'd even go another step further:
Proposed workflow:

1. Open Jamulus (code doesn't check  if all the sound setup is ok) 
2. After it has opened and it is possible to open settings, show the own slider; output audio (loop back mode) or give warnings e.g. that the sample rate is wrong (and of course how to fix it if you use ASIO4ALL),...

**Reply by gilgongo** [reply]
> The value is that you can try out the application in "dry run" mode

@dakhubgit That's this, isn't it? https://github.com/jamulussoftware/jamulus/issues/882

**Reply by dakhubgit** [reply]
Pretty much so, variant 2 in the issue-starting proposal.

**Reply by Hk1020** [reply]
> 
> > BTW has this got anything to do with the name of this discussion? I'm not complaining - it's just a discussion - but I was curious in case there was a connection I'd missed.
> 
> The connection with the discussion was tangential: in case of another different feature, namely being able to use a single client for transmitting several channels, you'd want an obvious ability to give your own sliders separate different profiles. **Being able to click on them for entering stuff would be easily discoverable** (actually also when only having a single slider: took me some time till I figured out where to configure it and it did not help that under XFCE4, Qt chose not to offer the respective menus at all).

Exactly. I had to tell several people how to set their name. Why hide this in some menu in the menu bar. Double click on one’s name and you can edit it.

**Reply by pljones** [reply]
If 
> b) We e.g. show the server list in disconnected state.

means _rather than_ the mixer window we show the server list, this makes some sense -- but the "settings" button needs to be available from there.

Then when started without an automatic connection, Jamulus would show the server list, encouraging the user to find somewhere to start jamming.  Once connected, the mixer would show - controlling the online session.  When disconnected, it goes away again.  But you can still get to Settings to set up.

Then it's a matter of not having Jamulus start up with the sound card open, which would allow the driver to be selected and set up.

**Comment by dakhubgit** [comment]
Peter L Jones <notifications@github.com> writes:

> If 
>> b) We e.g. show the server list in disconnected state.
>
> means _rather than_ the mixer window we show the server list, this
> makes some sense -- but the "settings" button needs to be available
> from there.
>
> Then when started without an automatic connection, Jamulus would show
> the server list, encouraging the user to find somewhere to start
> jamming.  Once connected, the mixer would show - controlling the
> online session.  When disconnected, it goes away again.  But you can
> still get to Settings to set up.
>
> Then it's a matter of not having Jamulus start up with the sound card
> open, which would allow the driver to be selected and set up.

Maybe it's the server list that should contain an extra button "Connect
to self" then?

-- 
David Kastrup

**Reply by pljones** [reply]
The client isn't a server, though, so that wouldn't make any sense at all.

Or do you mean a button that would spawn a server running locally and connect to that?

**Comment by dakhubgit** [comment]
Peter L Jones <notifications@github.com> writes:

> The client isn't a server, though, so that wouldn't make any sense at all.

Since when is a short-circuit not a connection?

-- 
David Kastrup

**Reply by pljones** [reply]
I'm sorry, that sentence makes no sense to me.  Can you translate it, please?

**Reply by dcorson-ticino-com** [reply]
Peter, I think you read that too quickly.
>Maybe it's the server list that should contain an extra button "Connect
to self" then?
I think with "server list" David is talking about the Connect window as a place to put a button to enable local loop-back.
It is, in any case, a possibility.

**Reply by pljones** [reply]
I think this would make for a very uncomfortable and confusing user experience, to be honest.

If you're on the mixer window, you need to be connected to a server.  That's simple to understand.

If you're on the connect screen, you're choosing a server to connect to.  That's simple to understand.

If you're trying to mix those two in different ways, that's not simple to explain or understand.

**Reply by dcorson-ticino-com** [reply]
Despite trying to think this through several times I am still not at all clear how this local loop-back should be presented and where the logical place is to enable it (and even how to name it so the neophyte understands).
I see this as a different case than the "mute myself" which is apparently loop-back while connected to a server.

This "Sound Setup" is the place to select drivers, set the input level and be sure there is no feedback, while not connected to a server.  The can of worms is that the input levels should be set according to the input meters and not the sound level heard at the output.  This is not intuitive especially if one has ones headphones on.  Having a fader to change the sound level in the headphones could be misconstrued as setting the input level, but if there is none the output levels could be uncomfortable leading people to wrongly reduce the input level.

Still looking for a solution to this problem other than having people read and understand directions, which I know will not happen.

**Comment by pljones** [comment]
> I see this as a different case than the "mute myself" which is apparently loop-back while connected to a server.

It's not "loop back".  It's more like being able to play along to a recording -- when the recording is playing, fine, but if you press stop, you can't play along any more.  It doesn't entirely bypass the process.  It needs to be receiving the audio to have something to mix in your local sound with.  If you want to play without the recording, you'd have to unplug from the recorder entirely.

**Reply by dakhubgit** [reply]
At some point of time you need to decide whether you want to view things from an implementation detail perspective or the practical meaning to the user when deciding what feature to implement and where.  It seems sort of arbitrary just when you declare one to trump the other.

**Reply by pljones** [reply]
No, it's a matter of hours in the day to get things done.  Completely rewriting Jamulus is going to take time that could be better spent on other improvements.

**Reply by dakhubgit** [reply]
"I could never get my choir/ensemble/orchestra to use a software as cumbersome and puzzling to use as that, so I'll better invest my programming skills and energy into a different project" does not really put the potentially available manhours in the day to best utilisation.  Working on UI issues sure comes at a cost but that does not mean that there is zero return.

**Reply by pljones** [reply]
You're not talking about a UI issue here.  Changing how the audio system works in Jamulus is not part of the UI.

**Comment by pljones** [comment]
> This "Sound Setup" is the place to select drivers, set the input level and be sure there is no feedback, while not connected to a server.

I was thinking this myself.  Split up the "Settings" window.  Have a separate "Sound set up" that lets you disconnect from the audio engine to allow the driver to be changed and set up (which is needed - you can't do that whilst the engine is running), and then has the input levels shown, without any level controls - but a hint to say "use your hardware input level to ensure the level here is right".  This is where the "Audio Channels" setting would go, too -- it's currently apparently unrelated as "Misc" but the input channel mapping is related.

The network settings, jitter buffer, audio quality and monitoring would go on a "Network Settings" page.

That would leave "Misc" with Audio Quality (which is arguably "Network Settings"), "New Client Level", "Skin" and "Language" -- and the current additional directory server (which I'd like moved to the connection window...).

**Comment by pljones** [comment]
"New Client Level" could move to a "Mixer" menu, along with "Load Mixer Channels Setup" and "Save Mixer Channels Setup".

"Skin" and "Language" could go to the "Profile" settings (which is a settings page).

All of that argues for either a tabbed settings page or a Settings menu to access each page.  It seems to be called "View" at the moment -- but has "Chat" on, too... which isn't settings...  (This is straying into the menu chaos ticket -- but such is the nature of discussing the UI...)

**Reply by dcorson-ticino-com** [reply]
Please look at my NewUI-02.  There I have already implemented a tabbed settings menu, which, of course, could still be improved.

**Reply by pljones** [reply]
Sorry, could you repost a link, please - there are over a thousand issues/discussions and it's hard to track them all.

**Reply by dcorson-ticino-com** [reply]
https://github.com/jamulussoftware/jamulus/discussions/1150
